lib/dns.js

```diff
@@ -38,6 +37,9 @@ function errnoException(err, syscall, hostname) {
   return ex;
 }
 
+function validIpVersion(isIpResult) {
+  return isIpResult !== 0;
```

Comment:
The documentation for `net.isIP()` describes the valid return values (including 0), so it shouldn't be private knowledge.